### PR TITLE
Implement Tunnel Configuration Check Feature

### PR DESCRIPTION
**Pull request by KristijanMitrik**

### CODE REVIEW

File: .vscode/launch.json
[the remark this answers is not in the document]
Comment (KristijanMitrik) [reply .vscode/launch.json:1]: Yes, I created it. So it's easier to run the app :)

**Review by dmacan23 — APPROVED** [review]

File: darwin/Classes/WireguardDartPlugin.swift [hunk 1]
```diff
@@ -1,109 +1,202 @@
...
+                result(
+                    nativeFlutterError(
+                        message: "tunnel not initialized, missing 'vpnManager'")
+                )
                 return
             }
             Logger.main.debug("Connection configuration: \(cfg)")
             Task {
                 do {
+                    mgr.isEnabled = true
```
Comment (Waldz) [inline darwin/Classes/WireguardDartPlugin.swift:116]: What is the point to do check before connect if we have separate methods to check before it?

Comment (KristijanMitrik) [reply darwin/Classes/WireguardDartPlugin.swift:116]: In case the tunnel had gotten disabled, re-enable and save it, then connect.

Comment (Waldz) [reply darwin/Classes/WireguardDartPlugin.swift]: But it should as the pseudo code on application side:

```
if ! checkTunnelConfiguration() {
   setupTunnel()
}

connect()
```

File: windows/connection_status_observer.cpp [hunk 2]
```diff
@@ -82,7 +96,6 @@ void CALLBACK ConnectionStatusObserver::ServiceNotifyCallback(void* ptr) {
 std::unique_ptr<flutter::StreamHandlerError<flutter::EncodableValue>> ConnectionStatusObserver::OnListenInternal(
     const flutter::EncodableValue* arguments, std::unique_ptr<flutter::EventSink<flutter::EncodableValue>>&& events) {
   sink_ = std::move(events);
-  // sink_->Success(flutter::EncodableValue(ConnectionStatusToString(ConnectionStatus::disconnected)));
```
Comment (Waldz) [inline windows/connection_status_observer.cpp:85]: Leftover?

**Review by Waldz — APPROVED** [review]

File: windows/connection_status_observer.cpp (hunk 2 above)
Comment (KristijanMitrik) [reply windows/connection_status_observer.cpp:85]: Yes

**Pull request merged**